### PR TITLE
Update CartoonSphere1D creator to use ZernikeB1 and internal BCs

### PR DESCRIPTION
**Pull request by michaeldmurphy1**
## Proposed changes

Identical cartoon-type boundary condition handling as in #7260

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.
- [ ] If a coding agent is used, have one of
      "Co-Authored-By: Claude Sonnet 4.6 <noreply@anthropic.com>",
      "Co-Authored-by: Codex <noreply@openai.com>", or
      "Co-Authored-By: GitHub Copilot CLI <noreply@microsoft.com>"
      as the last line of the commit, depending on the agent.

### CODE REVIEW

File: src/Domain/Creators/CartoonSphere1D.hpp [hunk 1]
```diff
@@ -208,3 +225,124 @@ class CartoonSphere1D : public DomainCreator<3> {
...
+  using RadialDistributions =
+      domain::creators::CartoonSphere1D::RadialDistributions;
+
+  static constexpr Options::String help = {"CartoonSphere1DOptionsHelper"};
+
+  // Default constructor required by Options system
+  CartoonSphere1DOptionsHelper() = default;
+
+  // Does not take cartoon BC
+  CartoonSphere1DOptionsHelper(
```
Comment (nilsdeppe) [inline src/Domain/Creators/CartoonSphere1D.hpp:263]: I missed this on the other PR, but I think this constructor could be defined in the cpp file, right?

File: src/Domain/Creators/CartoonSphere1D.hpp [hunk 2]
```diff
@@ -208,3 +225,124 @@ class CartoonSphere1D : public DomainCreator<3> {
...
+  Options::Context context_{};
+};
+}  // namespace domain::creators::detail
+
+// Options parsing specialization to automate CartoonSphere1D's cartoon
+// boundary condition
+template <>
+struct Options::create_from_yaml<domain::creators::CartoonSphere1D> {
+  template <typename Metavariables>
+  static domain::creators::CartoonSphere1D create(
```
Comment (nilsdeppe) [inline src/Domain/Creators/CartoonSphere1D.hpp:305]: Note to self: this must stay in the hpp since it depends on the metavariables.

File: tests/Unit/Domain/Creators/Test_CartoonSphere1D.cpp [hunk 3]
```diff
@@ -1,6 +1,7 @@
 // Distributed under the MIT License.
 // See LICENSE.txt for details.
 
+#include "Domain/Creators/OptionTags.hpp"
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Creators/Test_CartoonSphere1D.cpp]: move down

File: tests/Unit/Domain/Creators/Test_CartoonSphere1D.cpp [hunk 4]
```diff
@@ -37,57 +38,18 @@ create_boundary_condition(const bool outer) {
       outer ? Direction<3>::upper_xi() : Direction<3>::lower_xi(), 50);
 }
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Creators/Test_CartoonSphere1D.cpp:40]: Is there even a test for the new inner_radius==0 code?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

File: src/Domain/Creators/CartoonSphere1D.cpp
Comment (nilsdeppe) [inline src/Domain/Creators/CartoonSphere1D.cpp:29]: `Utilities/Gsl.hpp`

File: src/Domain/Creators/CartoonSphere1D.cpp [hunk 5]
```diff
@@ -120,7 +149,7 @@ CartoonSphere1D::CartoonSphere1D(
   for (size_t block = 0; block < num_blocks_; ++block) {
     const std::string block_name = "Block" + std::to_string(block);
     block_names_.emplace_back(block_name);
-    block_groups_[block_name].insert(block_name);
+    block_groups_["PositiveBlocks"].insert(block_name);
```
Comment (nilsdeppe) [inline src/Domain/Creators/CartoonSphere1D.cpp:177]: If you're changing the block (gorup) names, does the hpp dox need updating? I think we only sometimes document group names

File: src/Domain/Creators/CartoonSphere1D.hpp [hunk 6]
```diff
@@ -208,3 +225,124 @@ class CartoonSphere1D : public DomainCreator<3> {
 };
 
 }  // namespace domain::creators
+
+namespace domain::creators::detail {
+/// \brief Helper struct for CartoonSphere1D options parsing so the internal
+/// cartoon boundary condition at $x = 0$ is not a required argument.
+///
+/// \detail To get the cartoon-type boundary condition from a system we need
```
Comment (nilsdeppe) [inline src/Domain/Creators/CartoonSphere1D.hpp]: `\details`?

File: tests/Unit/Domain/Creators/Test_CartoonSphere1D.cpp [hunk 7]
```diff
@@ -37,57 +38,18 @@ create_boundary_condition(const bool outer) {
...
+                              TestCartoonBoundaryCondition<3>>();
+}
+
 template <typename T>
 std::string stringize(const std::vector<T>& t) {
   std::string result = get_output(t).replace(0, 1, "[");
   return result.replace(result.length() - 1, 1, "]");
 }
 
-std::string option_string(
```
Comment (nilsdeppe) [inline tests/Unit/Domain/Creators/Test_CartoonSphere1D.cpp:46]: Could you confirm that time-dependent options are still tested? Same with all the other things this handles

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
LGTM, please rebase and squash

**Review by nilsdeppe — APPROVED** [review]
